### PR TITLE
Minor refactor to signal handlers to remove potential sleeps

### PR DESCRIPTION
**Pull request by spookydonut**
## About The Pull Request
These are legacy sleepers, the remaining 12 are non-trivial to fix.

### CODE REVIEW

**Comment by Rohesie** [comment]
I suggest this is reviewed (and preferably tested) well. Last time the sleeping procs were just substituted with inovke async it broke telekinesis, made me refactor the attack chain, and had to deal with other stuff there.
Some things rely on the sleeps guarantees certain conditions further down the line haven't happened yet.
Invoke async is not necessarily a safe replacement.

File: code/datums/components/religious_tool.dm [hunk 1]
```diff
@@ -51,60 +51,22 @@
...
-		performing_rite = new selection2type(parent)
-		if(!performing_rite.perform_rite(user, parent))
-			QDEL_NULL(performing_rite)
-		else
-			performing_rite.invoke_effect(user, parent)
-			easy_access_sect.adjust_favor(-performing_rite.favor_cost)
-			QDEL_NULL(performing_rite)
-		return
+		INVOKE_ASYNC(src, .proc/perform_rite, user)
+		return (force_catalyst_afterattack ? null : COMPONENT_NO_AFTERATTACK)
```
Comment (Qustinnus) [inline code/datums/components/religious_tool.dm:69]:
```suggestion
		return (force_catalyst_afterattack ? NONE : COMPONENT_NO_AFTERATTACK)
```

isn't it better to return this as it expects a flag?

**Review by ghost — APPROVED** [review]
Looks good to me, bless you for doing the work noone else wants to do.

**Comment by optimumtact** [comment]
We'll test it live

**Pull request merged**